Sming/system/uart.cpp

```diff
@@ -156,23 +156,27 @@ void uart_stop_isr(uart_t* uart)
 }
 
 
-void uart_write_char(uart_t* uart, char c)
+size_t uart_write_char(uart_t* uart, char c)
 {
     if(uart == NULL || !uart->tx_enabled) {
-        return;
+        return 0;
     }
     while((USS(uart->uart_nr) >> USTXC) >= 0x7f);
     USF(uart->uart_nr) = c;
+    return 1;
 }
 
-void uart_write(uart_t* uart, const char* buf, size_t size)
+size_t uart_write(uart_t* uart, const void* buf, size_t size)
```

Comment:
What is the reason behind changing the `const char* buf` to `const void* buf` and casting it after that? Is there any useful use-case for it?

Reply:
Because serial port can be used for non-ascii data it means caller doesn't need to cast. Also no cast required to deal with signed/unsigned char arrays.

For example:

struct SomethingStruct mydata = {.... };
uart_write(uart, &mydata, sizeof(mydata)); // No cast required